### DISCUSSION TITLE
Allow focusable disabled buttons

### DISCUSSION THREAD

**Discussion by scottanthonymurray** [indiana-university/rivet-source#418] [Feature and enhancement requests]
Sometimes, activating a button in focus will cause the button itself to be disabled. For example, imagine you have previous/next buttons for pagination, and you want the buttons to conditionally disable on the border pages. As in, you don't want to go next from the last page or go previous on the first page. Activating the button under these conditions will disable itself.

What happens next with focus is inconsistent among browsers. Sometimes, the next focusable element is the next element at the point in which the focused button got disabled; this seems like the expected behavior, but it is actually against spec. Other times, focus goes to the top of the document; this is how IE/Edge works.
- https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/9548332/
- https://allyjs.io/tutorials/mutating-active-element.html#the-results

We encountered this same problem with the ESS Component Library, and we solved it by allowing disabled buttons to be focusable. As in, don't the use `disabled` attribute, but still make use of the same affordances as a button with the `disabled` attribute.
- https://github.iu.edu/ess-ux/components/issues/124
- https://github.iu.edu/ess-ux/components/wiki/Button#focusable-disabled-button

The concept of an element being disabled means it must be perceived as being disabled and behave as if it is disabled. Adding a `aria-disabled="true"` attribute makes is audibly perceived disabled to screen readers. Explicitly adding a `.rvt-button--disabled` modifier (or just implicitly by a `[aria-disabled="true"]` selector) could make it visually perceived disabled. And not responding to events (by way of how it's programmed) makes it behave as if it is disabled.
- https://www.w3.org/TR/wai-aria/states_and_properties#aria-disabled
- https://www.w3.org/TR/html5/forms.html#concept-fe-disabled

There is precedent for disabled elements to receive focus.
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Disabled_controls

**Comment by zttodd** [comment]
This should be included in a future release when re-evaluating the button component.
